opencodeblocks/blocks/codeblock.py

```diff
@@ -79,21 +81,102 @@ def init_run_button(self):
         """Initialize the run button"""
         run_button = QPushButton(">", self.root)
         run_button.move(int(self.edge_size), int(self.edge_size / 2))
-        run_button.setFixedSize(int(3 * self.edge_size), int(3 * self.edge_size))
-        run_button.clicked.connect(self.run_code)
+        run_button.setFixedSize(int(3 * self.edge_size),
+                                int(3 * self.edge_size))
+        run_button.clicked.connect(self.run_left_without_args)
         return run_button
 
+    def init_run_all_button(self):
+        """ Initialize the run all button """
+        run_all_button = QPushButton(">>", self.root)
+        run_all_button.setMinimumWidth(int(self.edge_size))
+        run_all_button.clicked.connect(self.run_right)
+        run_all_button.raise_()
+
+        return run_all_button
+
     def run_code(self):
         """Run the code in the block"""
-        # Erase previous output
-        self.stdout = ""
+
+        # Reset stdout
         self._cached_stdout = ""
-        self.source = self.source_editor.text()
-        # Create a worker to handle execution
-        worker = Worker(self.source_editor.kernel, self.source)
-        worker.signals.stdout.connect(self.handle_stdout)
-        worker.signals.image.connect(self.handle_image)
-        self.source_editor.threadpool.start(worker)
+
+        # Set button text to ...
+        self.run_button.setText("...")
+        self.run_all_button.setText("...")
+
+        # Run code by adding to code to queue
+        code = self.source_editor.text()
+        self.source = code
+        kernel = self.source_editor.kernel
+        kernel.execution_queue.append((self, code))
+        if kernel.busy == False:
+            kernel.run_queue()
+
+    def reset_buttons(self):
+        """Reset the text of the run buttons"""
+        self.run_button.setText(">")
+        self.run_all_button.setText(">>")
+
+    def check_input(self):
+        """ Checks wether a block has connected input blocks """
+        for input_socket in self.sockets_in:
+            for _ in input_socket.edges:
```

Comment:
Use if when for is not needed
```suggestion
            if len(input_socket.edges) > 0:
```